documentation/source/glossary.rst

```diff
@@ -0,0 +1,66 @@
+.. _glossary:
+
+Glossary
+========
+
+.. glossary::
+
+   async function
+      FIXME
+
+   coroutine function
+      The definition of a function that, when called, returns a coroutine object.
+      Implemented using :term:`async functions<async function>`.
```

Comment:
I would link back to Python's documentation on async syntax. Here is Python's glossary on [coroutines and coroutine functions](https://docs.python.org/3/glossary.html#term-coroutine) and [async def functions](https://docs.python.org/3/reference/compound_stmts.html#async-def).